### PR TITLE
Use TypedArray to achieve good performance

### LINKED ISSUE #191: Make sure large data are represented in TypedArray, and also make sure there is no deep copy
Purpose:
 - Make sure the data is represented in `TypedArray` (instead of JavaScript `Array`)
 - No deep copy when copying messages

e.g. in `sensor_msgs/msg/Image.msg`

```
uint8[] data          # actual matrix data, size is (step * rows)
```

### LINKED ISSUE #220: Make It Efficient: Primitive Type ROS Array
We should use direct `type* buffer` for primitive type ROS array bases on the below findings.

image

### PR DESCRIPTION
**Pull request by kenny-y**
Fix #191 
Fix #220

### CODE REVIEW

**Comment by kenny-y** [comment]
Known issues were logged as #232 #233 #234 #235

File: src/rcl_bindings.cpp [hunk 1]
```diff
@@ -696,6 +696,10 @@ NAN_METHOD(InitString) {
     info.GetReturnValue().Set(Nan::Undefined());
 }
 
+inline char* get_addr(v8::Local<v8::Value> buf) {
```
Comment (minggangw) [inline src/rcl_bindings.cpp]: As `get_addr` doesn't return a attribute of a class, like set/get pair, shall we use `GetAddress` to make the convention consistent. Others are good to me.

**Comment by kenny-y** [comment]
@qiuzhong what do you think of the changes of the tests? :)

**Review by minggangw — APPROVED** [review]

**Comment by qiuzhong** [comment]
@kenny-y , the changes are OK for me. But I would suggest you to put your added code of `test-array.js` in another test file because they're testing differently aspects. My code mainly focus on array data communication and your code is focus on array data verification.

**Pull request merged**